core/src/main/java/io/grpc/ServiceDescriptor.java

```diff
@@ -67,4 +81,13 @@ public String getName() {
   public Collection<MethodDescriptor<?, ?>> getMethods() {
     return methods;
   }
+
+  /**
+   * The generated code may attach an object to a service descriptor, such as the proto codegen
+   * attaching a object that allows retrieving the underlying proto object.
+   */
+  @Nullable
+  public Object getAttachedObject() {
```

Comment:
"attached object" is very, very vague. Vague is okay, but I'm torn to find something better. I think "object" is the part that is bothering me most. getAttachment? getSupplementalDescriptor? getRawDescriptor? getMarshallerDescriptor? getReflectionAttachment?

Depending on what this is, we may have a better name for `ProtoFileDescriptorWrapper` (e.g., `ProtoSupplementalDescriptor`).

Meh.